### ISSUE TITLE
schema migration and small files/lag

### ISSUE THREAD

**Issue by jmueller7** [confluentinc/kafka-connect-storage-cloud#412]
Using the Amazon S3 Connector.   When we update our schema to a new version (adding an optional field - schema registry compatibility is set to FORWARD) and then deploy our app across multiple servers, we go from having ~60MB avro files written every 60seconds, to many small ~10-20KB files written every second.  This then creates catastrophic lag.  

So, over this deploy time, there are versions of our app writing v1 and v2 versions of the schema at the same time.   The connector is seemingly writing out the files every time it sees a record with a different schema (v1->v2, v2->v1).  

Is this expected behavior?  We would expect that with schema compatibility set in the config, it would be smart enough to figure out how to continue to write just one file?  Are we missing something in our config?  Thanks for any help/advice!

Our config for the connector is as follows:
{
    "connector.class": "io.confluent.connect.s3.S3SinkConnector",
    "s3.region": "us-east-2",
    "partition.duration.ms": "60000",
    "topics.dir": "logging",
    "schema.compatibility": "FORWARD",
    "flush.size": "100000000",
    "topics": "log",
    "tasks.max": "24",
    "timezone": "UTC",
    "locale": "US",
    "s3.object.tagging": "true",
    "format.class": "io.confluent.connect.s3.format.avro.AvroFormat",
    "s3.part.retries": "9",
    "record.field": "timestamp",
    "partitioner.class": "io.confluent.connect.storage.partitioner.TimeBasedPartitioner",
    "schema.generator.class": "io.confluent.connect.storage.hive.schema.DefaultSchemaGenerator",
    "name": "s3-sink",
    "storage.class": "io.confluent.connect.s3.storage.S3Storage",
    "rotate.schedule.interval.ms": "60000",
    "timestamp.extractor": "RecordField",
    "s3.bucket.name": "bucket",
    "s3.retry.backoff.ms": "1000",
    "path.format": "'dt'=YYYYMMddHH/'pod=xyz'"
}

**Comment by OneCricketeer** [comment]
> writing out the files every time it sees a record with a different schema (v1->v2, v2->v1).

> Is this expected behavior? 

Yes. Avro files contain the schema version that's written. A file flush occurs when a schema difference is seen; the data in each file is not "rolled forward/backward"
